Research/lva-node-app/app.js

```diff
@@ -0,0 +1,337 @@
+const WebSocket = require('ws');
+const EventHubReader = require('./eventhubreader.js');
+const iothub = require('azure-iothub');
+const bodyParser = require('body-parser');
+const { request, response } = require('express');
+const { convertIotHubToEventHubsConnectionString } = require('./iot-hub-connection-string.js');
+
+const app = express();
+const server= require('http').createServer(app);
+const wss = new WebSocket.Server({server});
+const Client = iothub.Client;
+
+app.use(bodyParser.urlencoded({extended: false}));
+app.use(bodyParser.json());
+app.use(express.static('public'))
+app.use(express.static('public/pages'))
+
+/**
+ * @global
+ */
+global.DEVICE_ID="";
+global.MODULE_ID="lvaEdge";
+global.IOTHUB_CONNECTION_STRING="";
+global.IOTHUB_ENDPOINT="";
+
+var graphInstances={};
+var graphTopologies={};
+var cameras={};
+var eventHubReader;
+
+const PORT=5000;
+
+/**
+ * For all following declarations of function(req, res):
+ * @param {XMLHttpRequest} req - request object
+ * @param {XMLHttpRequestResponseType} res - request response object
+ */
+
+// render the html pages. the '/<name>' is what the url extension will look like in the browser
+app.get('/', function (req, res) 
+{
+    res.render('index.html')
+})
+app.get('/mediagraph', function (req, res) 
+{
+    res.render('mediagraph.html')
+})
+app.get('/configuration', function (req, res) 
+{
+    res.render('configuration.html')
+})
+app.get('/cameras', function (req, res) 
+{
+    res.render('cameras.html')
+})
+app.get('/output', function (req, res) 
+{
+    res.render('output.html')
+})
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////// 
+// handle GET requests
+
+/**
+* send client global variables upon request 
+* @returns {void}
+*/
+app.get('/getglobals', function (req, res) 
+{
+  console.log("get globals get function!");
+  let globals = 
+  [
+    {
+      "name": "graphInstances",
+      "value": graphInstances
+    },
+    {
+      "name": "graphTopologies",
+      "value": graphTopologies
+    },
+    {
+      "name": "cameras",
+      "value": cameras
+    }
+  ];
+  res.send(JSON.stringify(globals));
+})
+
+/**
+* start sending live stream hub messages to client via websocket 
+* @returns {void}
+*/
+app.get('/hubMessages', function(req, res)
+{
+  if (!validCredentialsAreSet(res))
+  {
+    return;
+  }
+  else
+  {
+    receiveHubMessages();
+    //always close requests
+    res.send();
+  }  
+})
+
+/**
+ * stop receiving messages from iot hub
+ * @returns {void}
+ */
+app.get('/stopMessages', function(req, res)
+{
+  eventHubReader.stopReadMessage();
+  res.send();
+})
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// handle POST requests
+
+/**
+ * invoke LVA methods handler
+ * @returns {void}
+ */
+app.post('/runmethod', function (req, res) 
```

Comment:
why post and not put for all/most of these?

Reply:
changed set globals to PUT. The rest don't modify variables, they just call functions, so I use POST. The ones where I don't need to get or modify anything, I use POST. I use get when retrieving values (like getGlobals)

Reply:
Alright, fine with me. One other one to maybe take a look at in terms of verbs is using Get for stopMessages....

Reply:
thanks good catch. Changed to post!